src/task.c

```diff
@@ -395,6 +325,154 @@ task_thread_create(const char            *name,
    return STATUS_OK;
 }
 
+/******************************************************************************
+ * Registering/deregistering threads that were not started by the task
+ * system.
+ ******************************************************************************/
+
+/*
+ * task_register_thread(): Register this new thread with the task system.
+ *
+ * This is for use by threads created outside of the splinter system.
+ *
+ * Registration implies:
+ *  - Acquire a new thread ID (index) for this to-be-active thread
+ *  - Allocate scratch space for this thread, and track this space..
+ */
+platform_status
+task_register_thread(task_system *ts,
+                     uint64       scratch_size,
+                     const char  *file,
+                     const int    lineno,
+                     const char  *func)
+{
+   threadid thread_tid;
+
+   debug_code(thread_tid = platform_get_tid());
+   debug_assert(
+      thread_tid == INVALID_TID,
+      "[%s:%d::%s()] Attempt to register already registered thread %lu\n",
+      file,
+      lineno,
+      func,
+      thread_tid);
+
+   thread_tid = allocate_threadid(ts);
+   if (thread_tid == INVALID_TID) {
+      return STATUS_NO_SPACE;
+   }
+   platform_set_tid(thread_tid);
+
+   debug_assert(ts->thread_scratch[thread_tid] == NULL,
+                "Scratch space should not yet exist.");
```

Comment:
The older version of this message was raised under `platform_assert()`. For the same reasons given a few lines earlier, can't we always do this assertion check even for release builds?

Also, the older message printed out the calling function's info, too: file, lineno, func. I found that information useful while debugging. Can we extend this message to print those values too?

Also, it will certainly be helpful to print the offending `thread_tid` which ran into this situation.